transport/transport_test.go

```diff
@@ -1800,47 +1800,74 @@ func testFlowControlAccountCheck(t *testing.T, msgSize int, wc windowSizeConfig)
 	}
 	server.mu.Unlock()
 	ct := client.(*http2Client)
-	cstream, err := client.NewStream(context.Background(), &CallHdr{})
-	if err != nil {
-		t.Fatalf("Failed to create stream. Err: %v", err)
-	}
-	msg := make([]byte, msgSize)
-	buf := make([]byte, msgSize+5)
-	buf[0] = byte(0)
-	binary.BigEndian.PutUint32(buf[1:], uint32(msgSize))
-	copy(buf[5:], msg)
-	opts := Options{}
-	header := make([]byte, 5)
-	for i := 1; i <= 10; i++ {
-		if err := ct.Write(cstream, nil, buf, &opts); err != nil {
-			t.Fatalf("Error on client while writing message: %v", err)
-		}
-		if _, err := cstream.Read(header); err != nil {
-			t.Fatalf("Error on client while reading data frame header: %v", err)
-		}
-		sz := binary.BigEndian.Uint32(header[1:])
-		recvMsg := make([]byte, int(sz))
-		if _, err := cstream.Read(recvMsg); err != nil {
-			t.Fatalf("Error on client while reading data: %v", err)
-		}
-		if len(recvMsg) != len(msg) {
-			t.Fatalf("Length of message received by client: %v, want: %v", len(recvMsg), len(msg))
+	const numStreams = 10
+	clientStreams := make([]*Stream, numStreams)
+	for i := 0; i < numStreams; i++ {
+		var err error
+		clientStreams[i], err = client.NewStream(context.Background(), &CallHdr{})
+		if err != nil {
+			t.Fatalf("Failed to create stream. Err: %v", err)
 		}
 	}
-	var sstream *Stream
+	var wg sync.WaitGroup
+	// For each stream send pingpong messages to the server.
+	for _, stream := range clientStreams {
+		wg.Add(1)
+		stream := stream
+		go func(stream *Stream) {
+			defer wg.Done()
+			msg := make([]byte, msgSize)
+			buf := make([]byte, msgSize+5)
+			buf[0] = byte(0)
+			binary.BigEndian.PutUint32(buf[1:], uint32(msgSize))
+			copy(buf[5:], msg)
+			opts := Options{}
+			header := make([]byte, 5)
+			for i := 1; i <= 10; i++ {
+				if err := ct.Write(stream, nil, buf, &opts); err != nil {
+					t.Errorf("Error on client while writing message: %v", err)
+					return
+				}
+				if _, err := stream.Read(header); err != nil {
+					t.Errorf("Error on client while reading data frame header: %v", err)
+					return
+				}
+				sz := binary.BigEndian.Uint32(header[1:])
+				recvMsg := make([]byte, int(sz))
+				if _, err := stream.Read(recvMsg); err != nil {
+					t.Errorf("Error on client while reading data: %v", err)
+					return
+				}
+				if len(recvMsg) != len(msg) {
+					t.Errorf("Length of message received by client: %v, want: %v", len(recvMsg), len(msg))
+					return
+				}
+			}
+		}(stream)
```

Comment:
This _or_ the `stream := stream` is needed to avoid the iterator reassignment problem.  I'm OK either way, but please remove one.